File: src/main/java/com/footverse/wishlist/dto/AddWishlistItemRequest.java
package com.footverse.wishlist.dto;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;

/**
 * Payload for {@code POST /wishlist}. Validation follows validation-spec §8; that the product
 * exists is a business rule enforced by the service, not a field-level constraint, and adding a
 * product the caller has already wishlisted is an idempotent no-op rather than an error
 * (business-rules → Wishlist).
 *
 * @param productId required, the product to wishlist
 */
public record AddWishlistItemRequest(
        @NotNull @Positive Long productId) {
}

File: app/lib/features/wishlist/models/add_wishlist_item_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'add_wishlist_item_request.g.dart';

/// The write body for adding a product to the wishlist (dto-spec §13).
///
/// A duplicate add is an idempotent no-op on the server (business-rules →
/// Wishlist); this model only serializes the wire body.
@JsonSerializable(createFactory: false)
class AddWishlistItemRequest {
  const AddWishlistItemRequest({required this.productId});

  final int productId;

  Map<String, dynamic> toJson() => _$AddWishlistItemRequestToJson(this);
}
